### ISSUE TITLE
Set-KeyNameValue with empty Value not possible

### ISSUE THREAD

**Issue by dwettstein** [dfensgmbh/biz.dfch.PS.Appclusive.Client#86]
Updating an existing KNV on Appclusive without giving the current value is not possible, because the parameter is set as `Mandatory = $true`. As a consequence, the filter for the _Value_ is always set. However, when the current value of the KNV is too long the script runs into troubles because of the maximum URL length.
~~Furthermore, when a different _Value_ and `-CreateIfNotExist` is given, Appclusive just creates a new KNV with the same _Key_ and _Name_! This is strange since one assume that a _Key-Name_ combination is unique.~~ (see note below)

I suppose to either make the parameter `Value` non-mandatory or allow the empty string as input (`[AllowEmptyString()]`).

**Comment by dfch** [comment]
@dwettstein note: a Key, Name combination is definitely *not* guaranteed to be unique. The oppsoite is true: we can have multiple same Key/Name combinations that only differ in its value.

**Comment by dfch** [comment]
Note: the current maximum URL length is a restriction on the ReverseProxy and maybe WebServer. We can check if we can have altered these settings with the infrastructure teams.

Alternate approach: if the key/name/value combination is *that* long, we can only query for substrings of it. @rufer7 what do you think?

**Comment by dfch** [comment]
@dwettstein the underlying model of `KeyNameValue` is set to `Required` so the Cmdlet correctly states this value as `Required` as well. So it is actually not a bug. I removed the *bug* tag.

If you think it is a viable feature to have a `null` value a `Value` we should discuss this.

**Comment by rufer7** [comment]
> Alternate approach: if the key/name/value combination is *that* long, we can only query for substrings of it. @rufer7 what do you think?

@dfch good idea. @stevenpi will change the behaviour accordingly

**Comment by dwettstein** [comment]
@dfch you're right. The required value is not a bug. The bug is the value and URL length. However, why should it be useful to have multiple key/name combinations? Assume we have two equal key/name combination, but one has an empty value created by accident. Then I have to implement an evaluation which KNV to take.

Nevertheless, what I'm doing now is simply just remove every KNV I want to update and then re-create it. Otherwise I would have to update them all by hand in the GUI because of the URL length bug.

**Comment by rufer7** [comment]
@dwettstein a possible use case for multiple entries with same key and name:

OsType to Network mappings

```
key: biz.dfch.Appclusive.OSTypeToNetwork
name: RHEL6
value: Network001

key: biz.dfch.Appclusive.OSTypeToNetwork
name: RHEL6
value: Network002
```

**Comment by rufer7** [comment]
Resolved by PR #113. Will be delivered with next release (4.11.3)
